src/us/kbase/workspace/database/provenance/Contributor.java

```diff
@@ -0,0 +1,314 @@
+	 *
+	 * @return the contributor's name
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Gets the credit name associated with the contributor.
+	 *
+	 * @return the contributor's credit name, if present
+	 */
+	public Optional<String> getCreditName() {
+		return Optional.ofNullable(creditName);
+	}
+
+	/**
+	 * Gets the ID for the contributor.
+	 *
+	 * @return the contributor ID, if present
+	 */
+	public Optional<String> getContributorID() {
+		return Optional.ofNullable(contributorID);
+	}
+
+	/**
+	 * Gets the contributor's affiliations.
+	 *
+	 * @return the contributor's affiliations, if present.
+	 */
+	public List<Organization> getAffiliations() {
+		return Common.getList(affiliations);
+	}
+
+	/**
+	 * Gets the roles that the contributor filled.
+	 *
+	 * @return the contributor roles, if present.
+	 */
+	public List<ContributorRole> getContributorRoles() {
+		return Common.getList(contributorRoles);
+	}
+
+	/**
+	 * Gets the roles that the contributor filled.
+	 *
+	 * @return the contributor roles, if present.
+	 */
+	public List<String> getContributorRoleStrings() {
+		return Common.immutable(
+			Common.getList(contributorRoles).stream()
+			.map(ContributorRole::getPid)
+			.collect(Collectors.toList()));
+		}
+
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(affiliations, contributorID, contributorRoles, contributorType, name, creditName);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Contributor other = (Contributor) obj;
+		return Objects.equals(affiliations, other.affiliations)
+				&& Objects.equals(contributorID, other.contributorID)
+				&& Objects.equals(contributorRoles, other.contributorRoles)
+				&& Objects.equals(contributorType, other.contributorType)
+				&& Objects.equals(name, other.name)
+				&& Objects.equals(creditName, other.creditName);
+	}
+
+	/**
+	 * Gets a builder for a {@link Contributor}.
+	 *
+	 * @param contributorType
+	 *                the type of contributor.
+	 * @param name    contributor name
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final ContributorType contributorType, final String name) {
+		final List<String> errorList = new ArrayList<>();
+		if (contributorType == null) {
+			errorList.add("contributorType cannot be null");
+		}
+		return new Builder(contributorType, name, errorList);
+	}
+
+	/**
+	 * Gets a builder for a {@link Contributor}.
+	 *
+	 * @param contributorType
+	 *                the type of contributor, as a string.
+	 * @param name    contributor name
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final String contributorType, final String name) {
+
+		final List<String> errorList = new ArrayList<>();
+		ContributorType ct = null;
+		try {
+			ct = ContributorType.getType(contributorType);
+		} catch (Exception e) {
```

Comment:
```suggestion
		} catch (IllegalArgumentException e) {
```
I think that's the right type anyway

Reply:
:+1: